Class-19-(Angular-Lesson-2).md

```diff
@@ -0,0 +1,35 @@
+## In the prep work this week (Angular Lesson 2), the students learned:
+
+1. Structural directives shape the page by adding and removing elements.
+    1. Structural directives start with a *, such as ``*ngIf``, ``*ngFor``.
+    1. ``*ngIf`` adds or removes the host element and it's descendants.
+    1. Each iteration of ``*ngFor`` creates a copy of the host element and it's descendants.
+1. How to handle an event with a ``(eventName)`` directive and then execute a simple statement or function call.
+    1. ``(click)``, ``(keyup)``, ``(mouseover)``, etc.
+1. How to use template reference variables and a click handler to take in user input - Example: ``<input #name (click)="addThing(name)">``.
+
+## Large Group Time (Instructor Notes)
+
+### Announcements
+
+1. Graded assignment #6 is due soon after class 20.
+1. Provide an introduction to what students can expect in unit 2.
+1. Check with your class Candidate Engagement Manager for any other announcements.
+
+### Topics That Require More Attention
+
+1. Q & A as needed for the prep-work exercises.
+1. Discuss the relative scale/priority for the different Angular directives - components, structural directives, attribute directives.
+1. Remind the students that only one structural directive can be placed inside an HTML tag. With this in mind, ask how we would approach a situation where we want to use two of these directives?
+1. Live code examples for ``*ngFor`` and ``*ngIf``.
+    1. Discuss the difference between *hiding* HTML elements vs. *removing* them.
+1. If time permits, discuss commonly used events besides the ``keyup``, ``keydown``, ``click``, and ``mouseover`` presented in the reading.
+1. Studio intro.
+
+## Studio (TF Notes)
+
+1. If you do not know Angular well, you must ABSOLUTELY complete ALL of the prep work yourself! Be the coach your students need.
```

Comment:
I would rephrase this: "Remember to do the prep work if you are struggling with Angular yourself!"

Reply:
Rephrased.  Thanks.